sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1406,17 +1406,27 @@ bool gnrc_netif_ipv6_wait_for_global_address(gnrc_netif_t *netif,
 
     /* wait for global address */
     msg_t m;
+    time_t t_stop = ztimer_now(ZTIMER_MSEC) + timeout_ms;
     while (!has_global) {
+        /* stop if timeout reached */
+        if (ztimer_now(ZTIMER_MSEC) > t_stop) {
+            DEBUG_PUTS("gnrc_netif: timeout reached");
+            break;
+        }
+        /* waiting for any message */
         if (ztimer_msg_receive_timeout(ZTIMER_MSEC, &m, timeout_ms) < 0) {
             DEBUG_PUTS("gnrc_netif: timeout waiting for prefix");
             break;
         }
 
-        if (ipv6_addr_is_link_local(m.content.ptr)) {
-            DEBUG_PUTS("gnrc_netif: got link-local address");
-        } else {
-            DEBUG_PUTS("gnrc_netif: got global address");
-            has_global = true;
+        /* scan all interfaces */
+        for (;(netif = gnrc_netif_iter(netif));) {
```

Comment:
You can't expect `netif` to be always `NULL` here and the state of `netif` is still needed in the cleanup phase later on.

```suggestion
        if (netif) {
            _is_bus_msg(netif, &has_global);
        } else {
            gnrc_netif_t *n = NULL;
            while ((n = gnrc_netif_iter(n))) {
                _is_bus_msg(netif, &has_global);
            }
        }
```

```C
static void _is_bus_msg(gnrc_netif_t *netif, bool *has_global)
{
    msg_bus_t* bus = gnrc_netif_get_bus(netif, GNRC_NETIF_BUS_IPV6);
    if (msg_is_from_bus(bus, &m) && ipv6_addr_is_global(m.content.ptr)) {
        DEBUG_PUTS("gnrc_netif: got global address");
        *has_global = true;
    }
}

Reply:
Done